ProcessLib/SmallDeformationWithLIE/LocalAssembler/SmallDeformationLocalAssemblerMatrixNearFracture-impl.h

```diff
@@ -0,0 +1,228 @@
+        bool is_axially_symmetric,
+        unsigned const integration_order,
+        SmallDeformationProcessData<DisplacementDim>& process_data)
+    : SmallDeformationLocalAssemblerInterface(
+          n_variables * ShapeFunction::NPOINTS * DisplacementDim,
+          dofIndex_to_localIndex),
+      _process_data(process_data),
+      _integration_method(integration_order),
+      _shape_matrices(
+          initShapeMatrices<ShapeFunction, ShapeMatricesType,
+                            IntegrationMethod, DisplacementDim>(
+              e, is_axially_symmetric, _integration_method)),
+      _element(e)
+{
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+
+    _ip_data.reserve(n_integration_points);
+    _secondary_data.N.resize(n_integration_points);
+
+    for (unsigned ip = 0; ip < n_integration_points; ip++)
+    {
+        _ip_data.emplace_back(*_process_data._material);
+        auto& ip_data = _ip_data[ip];
+        auto const& sm = _shape_matrices[ip];
+        ip_data._detJ = sm.detJ;
+        ip_data._integralMeasure = sm.integralMeasure;
+        ip_data._b_matrices.resize(
+            KelvinVectorDimensions<DisplacementDim>::value,
+            ShapeFunction::NPOINTS * DisplacementDim);
+
+        auto const x_coord =
+            interpolateXCoordinate<ShapeFunction, ShapeMatricesType>(e,
+                                                                     sm.N);
+        LinearBMatrix::computeBMatrix<DisplacementDim,
+                                      ShapeFunction::NPOINTS>(
+            sm.dNdx, ip_data._b_matrices, is_axially_symmetric, sm.N,
+            x_coord);
+
+        ip_data._sigma.resize(KelvinVectorDimensions<DisplacementDim>::value);
+        ip_data._sigma_prev.resize(
+            KelvinVectorDimensions<DisplacementDim>::value);
+        ip_data._eps.resize(KelvinVectorDimensions<DisplacementDim>::value);
+        ip_data._eps_prev.resize(
+            KelvinVectorDimensions<DisplacementDim>::value);
+        ip_data._C.resize(KelvinVectorDimensions<DisplacementDim>::value,
+                          KelvinVectorDimensions<DisplacementDim>::value);
+
+        _secondary_data.N[ip] = sm.N;
+    }
+}
+
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          int DisplacementDim>
+void SmallDeformationLocalAssemblerMatrixNearFracture<ShapeFunction, IntegrationMethod,
+                                    DisplacementDim>::
+assembleWithJacobian(
+    double const t,
+    Eigen::VectorXd const& local_u,
+    Eigen::VectorXd& local_b,
+    Eigen::MatrixXd& local_J)
+{
+    assert (_element.getDimension() == DisplacementDim);
+
+    auto const n_dof_per_var = ShapeFunction::NPOINTS * DisplacementDim;
+
+    auto localRhs_ru = local_b.segment(0, n_dof_per_var);
+    auto localRhs_du = local_b.segment(n_dof_per_var, n_dof_per_var);
+
+    auto localA_uu = local_J.block(0, 0, n_dof_per_var, n_dof_per_var);
```

Comment:
Eigen provides template versions of block: `block<N, M>(...)`. Since the size is known at compile time, you could use them. Same for `segment` above
Furthermore there are variants like upperLeftPart (or similar) which might in some cases make the code a bit easier to read.